internal/merger/merger.go

```diff
@@ -290,3 +290,62 @@ func MergeObservationDates(
 	}
 	return main
 }
+
+// MergeStatVarSummary merges two StatVarSummary maps.
+func MergeStatVarSummary(primary, secondary map[string]*proto.StatVarSummary) map[string]*proto.StatVarSummary {
+	merged := map[string]*proto.StatVarSummary{}
```

Comment:
maybe check the empty "primary" and "secondary" cases first to avoid any merge operation. Same for MergeBulkVariableInfoResponse()

Reply:
I had given it a thought but responses could include `nil` infos. The merge removes them. So keeping this as-is.